src/controller/CHIPDeviceController.h

```diff
@@ -42,7 +42,14 @@
 #include <transport/TransportMgr.h>
 #include <transport/raw/UDP.h>
 
+#if CONFIG_NETWORK_LAYER_BLE
+#include <ble/BleLayer.h>
+#endif
+
 namespace chip {
+namespace Ble {
+class BleLayer;
```

Comment:
why is this forward declaration needed?

The include should be sufficient. I believe this is to be able to pass in a Ble layer when no ble layer is used ... that seems hackish. 

I also see the Init call taking 5 parameters already. How about creating a 'InitParams' class that has ifdef-guards on the BLE layer?

This way you build the init parameters depending on what is supported.